### PR TITLE
MISUV-638: BusinessListCYA controller and update the redirect url for…

### PR DESCRIPTION
**Pull request by scullythornton1**
… trade name and accounting method pages

### CODE REVIEW

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodController.scala [hunk 1]
```diff
@@ -78,7 +76,7 @@ class BusinessAccountingMethodController @Inject()(mcc: MessagesControllerCompon
   }
 
   def backUrl(): String =
-    routes.BusinessTradeNameController.show().url
+    routes.BusinessListCYAController.show().url
```
[the remark this answers is not in the document]
Comment (scullythornton1) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodController.scala:79]: no the flow should lead to CYA as the page after trade name suppose to be business address and then the CYA page , however as we have not built business address page so for now, after trade name page should be CYA.

[the remark this answers is not in the document]
Comment (chandraAndem) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessAccountingMethodController.scala:79]: ok

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAController.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      answers = model,
+      postAction = uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.controllers.routes.BusinessAccountingMethodController.show(),
+      backUrl = backUrl(),
+      implicitDateFormatter = dateFormatter)
+
+  def show(): Action[AnyContent] = Action.async { implicit request =>
+    authService.authorised() {
+      incomeTaxSubscriptionConnector.getAllSelfEmployments().map {
+        case Right(Some(dataModel)) => Ok(view(dataModel.model))
+        case Right(None) => throw new InternalServerException(s"[BusinessListCYAController][show] - No data")
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAController.scala:56]: If the user has hit this page with no business data Right(None), couldn't we just send them to the first page (business start date)? Might be better than them seeing technical difficulties

Comment (scullythornton1) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAController.scala:56]: done the changed

**Pull request merged**